SwiftUITask/SwiftUITask/AudioViewModel/AudioViewModel.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  AudioViewModel.swift
+//  SwiftUITask
+//
+//  Created by Алексей Барлетов on 01.05.2024.
+//
+
+import AVFoundation
+import SwiftUI
+
+final class AudioViewModel: ObservableObject {
+    
+    var listTrak: [InformationAudio] = [
+        .init(nameExecutor: "Skryptonite",
+              nameTrak: "Snegopad",
+              timeTrak: "03.40" ,
+              nameImageTrak: .winter,
+              nameFile: "skryptoniteSnegopad"),
+        .init(nameExecutor: "MiyaGi",
+              nameTrak: "New Day",
+              timeTrak: "03.12",
+              nameImageTrak: .newDay,
+              nameFile: "miyaGiNewDay"),]
+    
+    var nameExuctor = "Skryptonite"
+    var nameTrak = "Snegopad"
+    var timeTrak = "03.40"
+    var nameImageTrak: ImageResource = .winter
+    
+    @Published  var currentTime = "00.00"
+    @Published   var timeDuration = 00.00
+    @Published public var maxDuration = 0.0
+    private var currentTrakIndex = 0
+    private var player: AVAudioPlayer?
+    
+    public func play() {
+        playSong()
+        player?.play()
+    }
+    public func stop() {
+        player?.stop()
+    }
+    
+    public func setTime(value: Float) {
+        guard let time = TimeInterval(exactly: value) else { return }
+        player?.currentTime = time
+        player?.play()
+    }
+    
+    private func playSong() {
+        guard let audioPath = Bundle.main.path(forResource: listTrak[currentTrakIndex].nameFile, ofType: "mp3") else { return  }
+        do {
+            try player = AVAudioPlayer(contentsOf: URL(fileURLWithPath: audioPath))
+            maxDuration = player?.duration ?? 00
+        } catch {
+            print(error.localizedDescription)
+        }
+        Timer.scheduledTimer(timeInterval: 0.5, target: self, selector: #selector(setupFormatTimer), userInfo: nil, repeats: true)
+    }
+    
+    @objc private func setupFormatTimer() {
+        guard let player else { return }
+        let time = Int(player.currentTime)
+        let minuts = time / 60
+        let second = time % 60
+        currentTime = NSString(format: "%02D:%02D", minuts, second) as String
+        timeDuration = player.currentTime
+    }
+    
+    func rewindForward() {
```

Comment:
сначала публичные методы должны идти, потом приватные 
```suggestion
    func rewindForward() {
```